webapp/src/components/common/dialogs/ImportDialog.tsx

```diff
@@ -1,29 +1,47 @@
 import { useEffect, useState } from "react";
-import * as R from "ramda";
-import { Box, LinearProgress, Paper, Typography } from "@mui/material";
-import Dropzone, { type Accept } from "react-dropzone";
-import { useMountedState } from "react-use";
+import { Box, Button, LinearProgress, Paper, Typography } from "@mui/material";
+import { FileRejection, useDropzone, type Accept } from "react-dropzone";
 import { useTranslation } from "react-i18next";
 import BasicDialog, { BasicDialogProps } from "./BasicDialog";
+import { blue, grey } from "@mui/material/colors";
+import useEnqueueErrorSnackbar from "../../../hooks/useEnqueueErrorSnackbar";
+import { toError } from "../../../utils/fnUtils";
+import { enqueueSnackbar } from "notistack";
+import { PromiseAny } from "../../../utils/tsUtils";
 
-interface Props {
-  open: BasicDialogProps["open"];
-  title?: string;
+interface ImportDialogProps extends Omit<BasicDialogProps, "actions"> {
   dropzoneText?: string;
   accept?: Accept;
-  onClose: VoidFunction;
+  onCancel: VoidFunction;
   onImport: (
     file: File,
-    onUploadProgress: (progress: number) => void,
-  ) => Promise<void>;
+    setUploadProgress: (progress: number) => void,
+  ) => PromiseAny;
```

Comment:
any reason to force any here ? I think in most cases its a void Promise or undefined

Reply:
The server call can return a value (like `importFile` in 'study.ts')